lib/bolt/pal/yaml_plan.rb

```diff
@@ -60,15 +99,25 @@ def validate
         end
 
         @steps.each do |step|
+          validate_step_keys(step, step_number)
+
+          begin
+            step.each { |k, v| validate_puppet_code(k, v) }
+          rescue Bolt::Error => e
```

Comment:
I still need to think more about this. I want to be able to report `step_number` and `step['name']` in error message *in addition to* the key that does not parse with it's associated puppet parser error message. Instead of carrying the `step_number` and `step['name']` through every recursive call as a static parameters in the `validate_puppet_code`  method, I catch the puppet error, convert it to bolt error with the info I want and then can catch it and report here. I guess if I stick with this I should raise  a more specific error in `validate_puppet_code`.